internal/k8shandler/forwarding.go

```diff
@@ -47,6 +48,14 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 	op[generator.ClusterTLSProfileSpec] = tls.GetClusterTLSProfileSpec(tlsProfile)
 	EvaluateAnnotationsForEnabledCapabilities(clusterRequest.Forwarder, op)
 
+	// Use logcollector SA token/ca.crt for the legacy case
+	if clusterRequest.Forwarder.Spec.ServiceAccountName == constants.CollectorServiceAccountName {
```

Comment:
I would move this logic into ClusterRequest#SetOutputSecret so that all the "retrieve secret and create map" logic is in one place